processor/batchprocessor/batch_processor.go

```diff
@@ -171,9 +170,9 @@ func (bp *batchProcessor) Start(context.Context, component.Host) error {
 }
 
 // Shutdown is invoked during service shutdown.
-func (bp *batchProcessor) Shutdown(context.Context) error {
+func (bp *batchProcessor) Shutdown(ctx context.Context) error {
+	bp.shutdownCtx = ctx
```

Comment:
This context passed to Shutdown isn't going to be always cancelled. We shouldn't use it

Also, you are replacing the `bp.shutdownCtx` with another instance. So the old context passed to the downstream consumer is unchanged.